lms/djangoapps/bookmarks/services.py

```diff
@@ -31,7 +59,13 @@ def bookmarks(self, course_key):
         Returns:
             list of dict:
         """
-        return api.get_bookmarks(self._user, course_key=course_key, fields=DEFAULT_FIELDS + OPTIONAL_FIELDS)
+        bookmarks = api.get_bookmarks(self._user, course_key=course_key, fields=DEFAULT_FIELDS + OPTIONAL_FIELDS)
+
+        if course_key.run is not None:
```

Comment:
If `course_key.run is None`, what condition does that indicate? Why can't we cache in that case? What kind of performance losses do you get if you can't cache?